openmcyclus/DepleteReactor.py

```diff
@@ -8,8 +11,603 @@ class DepleteReactor(Facility):
+        if (self.n_assem_spent - self.spent_fuel.count) < npop:
+            #self.record("DISCHARGE", "failed")
+            return False
+
+        ss = str(npop) + " assemblies"
+        #self.record("DISCHARGE", ss)
+        print("time:", self.context.time, "discharge", ss)
+
+        core_pop = self.core.pop_n(npop)
+        for ii in range(len(core_pop)):
+            self.spent_fuel.push(core_pop[ii])
+        tot_spent = 0
+        
+        for ii in range(len(self.fuel_outcommods)):
+            spent_mats = self.peek_spent()
+            if self.fuel_outcommods[ii] in spent_mats:
+                mats = spent_mats[self.fuel_outcommods[ii]]
+                tot_spent += mats.quantity
+                lib.record_time_series("supply"+self.fuel_outcommods[ii], self, tot_spent)
+
+        return True
+
+    def load(self):
+        '''
+        Determine number of assemblies to load, either the 
+        number of assemblies needed for the core less the number 
+        already in the core or the number held by the 
+        fresh fuel inventory. If no assemblies are needed, then 
+        return. 
+
+        Record the number of assemblies that are to be loaded, then move 
+        them from the fresh fuel inventory to the core inventory. 
+        '''
+        n = min((self.n_assem_core-self.core.count), self.fresh_fuel.count)
+        print("time:", self.context.time, "load", n, "assemblies")
+        if n == 0:
+            return
+        ss = str(n) + " assemblies"
+        #self.record("LOAD", ss)
+        assemblies = self.fresh_fuel.pop_n(n)
+        for ii in range(len(assemblies)):
+            self.core.push(assemblies[ii])
+        return 
+
+    def transmute(self, n_assem):
+        '''
+        Get the material composition of the minimum of the 
+        number of assemblies specified or the number of 
+        assemblies in the core. 
+
+        If there are more assemblies in the core than what will 
+        be removed, then rotate the untransmuted materials to the back 
+        of the buffer.
+
+        Record the number of assemblies to be transmuted. Transmute the fuel
+        by changing the recipe of the material to that of the 
+        fuel_outrecipes
+
+        There seem to be two Transmute functions in the cycamore reactor?
+        '''
+        old = self.core.pop_n(min(n_assem, self.core.count))
+        for ii in range(len(old)):
+            self.core.push(old[ii]) 
+        ss = str(len(old)) + " assemblies"
+        #self.record("TRANSMUTE", ss)
+        print("time:", self.context.time, "transmute", ss)
+        for ii in range(len(old)):
+            print("call OpenMC")
+        return
+
+    def record(self, event, val):
+        '''
+        Record a reactor event to the output database with the 
+        given name and note val
+
+        Parameters:
+        -----------
+        event: str 
+            name of event
+        val: str
+            value of event
+        '''
```

Comment:
Will do. All of the parameters and outputs of these functions are never seen by the user and are passed between the archetype and the cyclus kernel, so that hasn't been a high priority.